crates/trie/trie/src/node_iter.rs

```diff
@@ -67,6 +67,10 @@ pub struct TrieNodeIter<C, H: HashedCursor> {
     /// The key that the [`HashedCursor`] previously advanced to using [`HashedCursor::next`].
     #[cfg(feature = "metrics")]
     previously_advanced_to_key: Option<B256>,
+    /// Stores the result of the last successful `next_hashed_entry`, used to avoid a redundant
+    /// `seek_hashed_entry` call if the walker points to the same key that was just returned by
+    /// `next()`.
```

Comment:
```suggestion
    /// Stores the result of the last successful [`Self::next_hashed_entry`], used to avoid a redundant
    /// [`Self::seek_hashed_entry`] call if the walker points to the same key that was just returned by
    /// `next()`.
```